static/app/components/events/interfaces/spans/spanBar.tsx

```diff
@@ -104,19 +104,26 @@ const INTERSECTION_THRESHOLDS: Array<number> = [
 
 export const MARGIN_LEFT = 0;
 
-type SpanBarProps = {
+export type SpanBarProps = {
+  addExpandedSpan: (span: Readonly<ProcessedSpanType>, callback?: () => void) => void;
```

Comment:
Since the list is virtualized, we need to manually keep track of spans which have been expanded with this map, since otherwise they will not retain their expansion if they are scrolled out of view.